src/channel.cr

```diff
@@ -114,6 +114,10 @@ class Channel(T)
     end
   end
 
+  # Closes the channel.
+  # The method prevents any new value from being sent to / received from the channel.
+  # It wakes up any sender / receiver fibers waiting on the channel.
+  # It only has effect *after* any previously enqueued value in the channel has been received, up to the channel's capacity.
```

Comment:
I appreciate the wording can be improved, so happy to take suggestions here

Reply:
I'd probably just say

> All items successfully sent to the channel can be received before `receive` considers the channel closed, but `send` will consider the channel closed before `close` returns.